### PR TITLE
OWASP MobileTop10

### PR DESCRIPTION
**Pull request by shaopeng-gh**

### CODE REVIEW

**Comment by eddynaka** [comment]
> # SARIF Taxonomies

let's wait yong's pr, so u can apply some changes here.

---
In reply to: [857586676](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857586676) [](http://example.com/codeflow?ancestors=857586676)

---
In reply to: [857586676](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857586676) [](http://example.com/codeflow?ancestors=857586676)

---
In reply to: [857586676](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857586676) [](http://example.com/codeflow?ancestors=857586676)

---
In reply to: [857586676](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857586676) [](http://example.com/codeflow?ancestors=857586676)

---
In reply to: [857586676](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857586676) [](http://example.com/codeflow?ancestors=857586676)

---
Refers to: README.md:1 in 10e693c. [](commit_id = 10e693c633f4aef657e8e60dfb589c21b8f11407, deletion_comment = False)

File: Tools/Taxonomy/Common/Constants.cs [hunk 1]
```diff
@@ -46,13 +46,27 @@ private static class Guid
...
+            Guid = Guid.OwaspASVSV402,
             Name = "OWASP",
             Location = "https://raw.githubusercontent.com/sarif-standard/taxonomies/main/OWASP_ASVS_v4.0.2.sarif"
         };
 
+        public static TaxonomyData OwaspMobileTop102014 = new TaxonomyData
+        {
+            Guid = Guid.OwaspMobileTop102014,
+            Name = "OWASP Mobile Top 10 2014",
+            Location = "https://owasp.org/www-project-mobile-top-10/"
```
Comment (eddynaka) [inline Tools/Taxonomy/Common/Constants.cs]: >https://owasp.org/www-project-mobile-top-10 [](http://example.com/codeflow?start=24&length=43)

this is not right #Closed

File: Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                },
+                Guid = version switch
+                {
+                    "2014" => Constants.OwaspMobileTop102014.Guid,
+                    "2016" => Constants.OwaspMobileTop102016.Guid,
+                    _ => throw new NotImplementedException()
+                },
+                InformationUri = new Uri(version switch
+                {
+                    "2014" => Constants.OwaspMobileTop102014.Location,
```
Comment (eddynaka) [inline Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs]: >Constants.OwaspMobileTop102014.Location [](http://example.com/codeflow?start=30&length=39)

normally, this is a hard-coded location #Closed

Comment (eddynaka) [reply Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs]: pls, check the other taxonomies

**Comment by shaopeng-gh** [comment]
> # SARIF Taxonomies

just merged from master, start working on the changes

---
In reply to: [857586676](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857586676) [](http://example.com/codeflow?ancestors=857586676,857586676)

---
Refers to: README.md:1 in 10e693c. [](commit_id = 10e693c633f4aef657e8e60dfb589c21b8f11407, deletion_comment = False)

File: Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs (hunk 2 above)
Comment (shaopeng-gh) [reply Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs]: yes I can see in Yong's PR this Location is the location of .sarif file not the location of the source.
Fixed.

File: Tools/Taxonomy/Common/Constants.cs (hunk 1 above)
Comment (shaopeng-gh) [reply Tools/Taxonomy/Common/Constants.cs]: fixed.

File: README.md [hunk 3]
```diff
@@ -33,6 +33,12 @@ Generate OWASP ASVS Sarif file
 generate-owasp --source-file-path "OWASP Application Security Verification Standard 4.0.2-en.csv" --target-file-path "OWASP_ASVS_v4.0.2.sarif" --version "4.0.2"
 ```
 
+Generate OWASP Mobile Top 10 Sarif file
+
+```bash
+generate-owasp --source-file-path "\www-project-mobile-top-10-master\2014-risks" --target-file-path "OWASP_MobileTop10_v2014.sarif" --version "2014" --release-date "2014-10-01"
```
Comment (eddynaka) [inline README.md]: >--release-date "2014-10-01" [](http://example.com/codeflow?start=149&length=27)

remove this #Closed

Comment (yongyan-gh) [inline README.md]: >generate-owasp [](http://example.com/codeflow?start=0&length=14)

is the argument "type" missing here? #Closed

File: README.md [hunk 4]
```diff
@@ -33,6 +33,12 @@ Generate OWASP ASVS Sarif file
 generate-owasp --source-file-path "OWASP Application Security Verification Standard 4.0.2-en.csv" --target-file-path "OWASP_ASVS_v4.0.2.sarif" --version "4.0.2"
```
Comment (yongyan-gh) [inline README.md]: >generate-owasp [](http://example.com/codeflow?start=0&length=14)

also missing type argument #Closed

File: Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                });
+            }
+
+            return records;
+        }
+
+        private Run ConvertToSarif(List<OwaspMobileTop10Record> records, string version)
+        {
+            IList<ToolComponent> taxonomies = new List<ToolComponent>();
+            ToolComponent toolComponent = new ToolComponent
```
Comment (yongyan-gh) [inline Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs:77]: >ToolComponent [](http://example.com/codeflow?start=46&length=13)

do we set release date here? #Closed

**Comment by shaopeng-gh** [comment]
> # SARIF Taxonomies

just done fix the changes

---
In reply to: [857999992](https://github.com/sarif-standard/taxonomies/pull/11#issuecomment-857999992) [](http://example.com/codeflow?ancestors=857999992,857586676,857586676)

---
Refers to: README.md:1 in 10e693c. [](commit_id = 10e693c633f4aef657e8e60dfb589c21b8f11407, deletion_comment = False)

File: README.md (hunk 4 above)
Comment (shaopeng-gh) [reply README.md]: yes, the previous commit was just the merge from master, just fixed

File: Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs (hunk 5 above)
Comment (shaopeng-gh) [reply Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs:77]: there is no release day in the website and I see the field is optional. so I did not set it.

**Review by eddynaka — APPROVED** [review]
:shipit:

**Pull request merged**